src/stacked_borrows/stack.rs

```diff
@@ -11,14 +11,79 @@ use std::ops::Range;
 #[cfg(feature = "stack-cache")]
 const CACHE_LEN: usize = 32;
 
+// We store the permission in the low bits of the tag, leaving only 62 bits for the tag. But that
+// should be more than enough.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+struct PackedItem(u64);
+
+impl PackedItem {
+    const TAG_MASK: u64 = u64::MAX << 2;
```

Comment:
I think the various uses of `2` in this impl should be using an associated const